src/prefect/core/flow.py

```diff
@@ -943,12 +951,17 @@ def _run(
             ## begin a single flow run
             while not flow_state.is_finished():
                 runner = runner_cls(flow=self)
+                task_ctxts = kwargs.pop("task_contexts", {})
+                for t in self.tasks:
+                    task_ctxts.setdefault(t, dict())
+                    task_ctxts[t].update(task_run_id=str(uuid.uuid4()))
```

Comment:
IIUC this mutates the external dicts passed in as `task_contexts`, which might lead to unexpected behavior:

```python
task_contexts = {task: {...}}
flow.run(task_contexts=task_contexts)
# task_contexts is now mutated from inside the `run` call
```

Perhaps:

```suggestion
                task_ctxts = kwargs.pop("task_contexts", {})
                for t in self.tasks:
                    ctx = task_ctxts.setdefault(t, {}).copy()
                    ctx.update(task_run_id=str(uuid.uuid4()))
```